shoop/admin/modules/orders/static_src/create/reducers/lines.js

```diff
@@ -27,30 +35,60 @@ function setLineProperties(linesState, lineId, props) {
         if (line.id === lineId) {
             return _.assign({}, line, props);
         }
+
         return line;
     });
 }
 
+function getDiscountsAndTotal(quantity, baseUnitPrice, unitPrice, updateUnitPrice=false) {
+    var updates = {};
+    if (updateUnitPrice) {
+        updates.unitPrice = unitPrice;
+    }
+    var totalBeforeDiscount = baseUnitPrice * quantity;
+    var total = (unitPrice * quantity).toFixed(2);
+    updates.total = total;
+    if (baseUnitPrice < unitPrice) {
+        updates.discountPercent = 0;
+        updates.discountAmount = 0;
+        return updates;
+    }
+    var discountAmount = totalBeforeDiscount - total;
+    updates.discountAmount = discountAmount;
+    updates.discountPercent = ((discountAmount / totalBeforeDiscount) * 100).toFixed(2);
+    return updates;
+}
+
 function updateLineFromProduct(state, {payload}) {
     const {id, product} = payload;
     const line = _.detect(state, (sLine) => sLine.id === id);
     if (!line) {
         return state;
     }
-    const updates = {};
+    var updates = {};
+    if (!product.sku) {
+        // error happened before getting actual product information
+        updates.errorText = product.errorText;
+        return setLineProperties(state, id, updates);
+    }
     if (line.unitPrice === 0) {
+        updates = getDiscountsAndTotal(product.quantity, product.baseUnitPrice.value, product.unitPrice.value);
+        updates.baseUnitPrice = product.baseUnitPrice.value;
         updates.unitPrice = product.unitPrice.value;
         updates.unitPriceIncludesTax = product.unitPrice.includesTax;
     }
     updates.sku = product.sku;
     updates.text = product.name;
+    updates.quantity = product.quantity;
+    updates.step = product.purchaseMultiple;
+    updates.errorText = product.errorText;
     return setLineProperties(state, id, updates);
 }
 
 function setLineProperty(state, {payload}) {
     const {id, property, value} = payload;
     const line = _.detect(state, (sLine) => sLine.id === id);
-    const updates = {};
+    var updates = {};
```

Comment:
Can't use const here since later on getDiscountAndTotals is assigned to this variable.